packages/i18nify-go/modules/currency/conversion_test.go

```diff
@@ -0,0 +1,84 @@
+package currency
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConvertCurrency(t *testing.T) {
+	tests := []struct {
+		name               string
+		currencyCode       string
+		amount             interface{}
+		minorExpectedValue float64
+		majorExpectedValue float64
+		expectedError      string
+	}{
+		{
+			name:               "Valid USD Conversion",
+			currencyCode:       "USD",
+			amount:             1234.0,
+			minorExpectedValue: 123400,
+			majorExpectedValue: 12.34,
+			expectedError:      "",
+		},
+		{
+			name:               "Valid EUR Conversion",
+			currencyCode:       "EUR",
+			amount:             4568,
+			minorExpectedValue: 456800,
+			majorExpectedValue: 45.68,
+			expectedError:      "",
+		},
+		{
+			name:               "Valid GBP Conversion",
+			currencyCode:       "GBP",
+			amount:             100.0,
+			minorExpectedValue: 10000.00,
+			majorExpectedValue: 1.00,
+			expectedError:      "",
+		},
+		{
+			name:               "Invalid Currency Code",
+			currencyCode:       "INR0",
+			amount:             100.0,
+			minorExpectedValue: 0,
+			majorExpectedValue: 0,
+			expectedError:      "currency code 'INR0' not found",
+		},
+		{
+			name:               "Negative Amount",
+			currencyCode:       "USD",
+			amount:             -100.0,
+			minorExpectedValue: 0,
+			majorExpectedValue: 0,
+			expectedError:      "amount cannot be negative",
+		},
```

Comment:
add other currencies as well.. especially the three decimal ones.

Reply:
added three as well as four decimal ones in test cases. Thanks